perceval/backend.py

```diff
@@ -581,6 +701,9 @@ def fetch(backend_class, backend_args, category, filter_classified=False,
     try:
         for item in items:
             yield item
+
+        summary = repr(backend.summary)
+        sys.stderr.write(summary)
```

Comment:
I don't really like this. We need to find a way to provide access  to the summary object from this function. Others will find this useful. For example, in Arthur, this function is called by workers. The workers generate a result that is quite similar to this summary object. Instead of generating a new object, it will be great if the object is the same.

I think we should create a new function for this purpose, something that can return a generator of items and the result. Not sure if with a callable we can do it.

Reply:
With the changes you made this wouldn't be needed anymore here. The code that will write the summary will be either in the command or in the `perceval` script. Probably the summary should be written in the log.